### PR TITLE
Feature/initial frontend styling

### PR DESCRIPTION
**Pull request by fraserclark**
## Description

This is a bit of a work in progress -as to not slow us down, I've PR'd my work so far. I think Ill need to take another pass at alot of this

The main changes are

Style guide with styled components for 
- Input
- Select(Aria)
- Radio buttons
- Checkbox & Checkbox group
- Button
- Initial cards (I stopped)
- Question Card
- Section Card
- Tabs 

Template Pages
In the building out of components, I need "real" stuff, so I built out boilerplate pages for the template builder
http://localhost:3000/template/tpl_2525235
http://localhost:3000/template/tpl_2525235/section/sec_123456
http://localhost:3000/template/tpl_2525235/q/q_abcdef

CSS Cleanup
I have tidied up some of the CSS files in relation to the components.

I think future work we need to really clean up the main CSS file as its pretty inconsistent might be something to discuss on Tuesday if we want to assign a task for that etc 

## Type of change
These are mostly non-breaking changes as front-end changes. 

## How Has This Been Tested?
Just in browsers; Not add tests for the pages as they are "placeholder/boilerplate"

I will go through entire checklist once you guys provide feedback etc - but at least you can start using some of the components etc

### CODE REVIEW

File: .gitignore
Comment (fraserclark) [inline .gitignore:1]: Im using phpstorm so added this to gitignore

File: app/layout.tsx
Comment (fraserclark) [inline app/layout.tsx:1]: Wireframes are in Inter - so TBC if we should swap out Poppins

File: app/template/[templateId]/page.tsx
Comment (fraserclark) [inline app/template/[templateId]/page.tsx:1]: One of the boilerplate pages discussed - can be entirely deleted - just need something to build against

File: styles/responsive.scss
Comment (fraserclark) [inline styles/responsive.scss:1]: Looks like the $xl breakpoint was missing so I added it

**Comment by jupiter007** [comment]
Thanks @fraserclark. I'll take a look. It says this PR is in Draft, rather than "Open". I think you should just put it in "Ready for review" and can respond to comments as needed within this PR.

File: components/SectionHeaderEdit/SectionHeaderEdit.scss [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+.edit-button, .order-button, .expand-button {
+  background: none;
+  border: none;
+  cursor: pointer;
+  padding: 0.3125rem 0.625rem;
+  margin-left: 0.625rem;
+  font-size: 0.875rem;
+  &:hover {
```
Comment (jupiter007) [inline components/SectionHeaderEdit/SectionHeaderEdit.scss:36]: L36 - Did you mean to leave the "hover" empty?

File: components/SubHeader/index.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import { faCloud, faEnvelope } from '@fortawesome/free-solid-svg-icons';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+//import { faCloud, faEnvelope } from '@fortawesome/free-solid-svg-icons';
```
Comment (jupiter007) [inline components/SubHeader/index.tsx:1]: L1,2 - You can just delete the commented out fontawesome icons

File: styles/_elements.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-
+/*
```
Comment (jupiter007) [inline styles/_elements.scss:1]: Is there a reason we are keeping the commented styles around?

File: styles/globals.scss
[the remark this answers is not in the document]
Comment (jupiter007) [reply styles/globals.scss:1]: @fraserclark I agree that we should be using "rems".

I remember that the styles in globals.scss are temporary fillers until we establish the final design/branding. So feel free to make changes and propose improvements.

File: styles/responsive.scss
Comment (jupiter007) [reply styles/responsive.scss:1]: Thanks for adding that.

**Comment by jupiter007** [comment]
@fraserclark thanks for the PR. It's looking great and you've made great progress.

I only had a few comments. The main things were that I was hoping we would be using CSS Modules for our components, and I saw some weird tabbing or indenting in the components.

None of the design/styling is final, from what I understand, so you should feel free to override styles based on what you think will be the final look and feel.

As I mentioned in another ticket, the Card component work was meant to be done as part of another ticket ([#83](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/83)). If you need to make more updates for that, you can do it in that ticket. So we can leave what you have so far for the card here, and just make any needed improvements to it in the other ticket if you want.

**Comment by jupiter007** [comment]
Oh, @fraserclark, I also forgot to ask you to update the CHANGELOG.md file. Just a short blurb about adding new components is fine.

**Comment by fraserclark** [comment]
Thanks @jupiter007 will work through your comments!

Cool, I will add a summary to the changelog!

Yep will update the Card ticket and also I sort of adjusted the tabs to be the real aria tabs too. So will update that tickets, and any future work will be in a separate PR requests for both to tie them to the work ticket

File: app/layout.tsx
Comment (fraserclark) [reply app/layout.tsx:1]: Cool lets make this a separate card to do the task itself

File: styles/globals.scss
[the remark this answers is not in the document]
Comment (fraserclark) [reply styles/globals.scss:1]: Will create a separate card for trying to align the temporary with the new

File: .gitattributes [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Enforce LF line endings for all text files
+* text=auto
```
[the remark this answers is not in the document]
Comment (fraserclark) [reply .gitattributes:2]: Probably not - but Andre & I had issues in the past with line endings - as im on windows and he was using linux, so was trying to get ahead of it a bit.

**Comment by briri** [comment]
looks like this one needs the latest from development merged in.

**Review by jupiter007 — APPROVED** [review]
Hi @fraserclark, I approved this PR so that the components will be available to build out the new pages. It looks like the only comment not addresses was adding onMoveUp() and onMoveDown() functionality to the SectionHeaderEditProps. I created a separate ticket for that and assigned it to you: https://github.com/CDLUC3/dmsp_frontend_prototype/issues/136

**Pull request merged**